src/v/kafka/server/handlers/describe_configs.cc

```diff
@@ -77,12 +79,124 @@ static ss::sstring describe_as_string(const T& t) {
     return ssx::sformat("{}", t);
 }
 
+template<typename T>
+consteval bool is_short() {
+    if constexpr (std::is_same_v<T, char16_t> || std::is_same_v<T, char8_t>) {
```

Comment:
it might be better just to use sizeof<T> here, check if the value is less than X + is_integral

Reply:
```
template<typename T>
constexpr bool is_short = std::is_integral<T>::value && sizeof(T) < 32;

template<typename T>
constexpr bool is_integer = std::is_integral<T>::value && sizeof(T) < 64; // Maybe 42? not sure

template<typename T>
constexpr bool is_long = std::is_integral<T>::value && sizeof(T) == 64; // same here, depends on definition of "long" for kafka API
```